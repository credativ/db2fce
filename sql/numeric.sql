-- INTEGER()/INT()
SELECT INTEGER(1.00) FROM SYSIBM.SYSDUMMY1;
SELECT INTEGER(1.50) FROM SYSIBM.SYSDUMMY1;
SELECT INTEGER(1.99) FROM SYSIBM.SYSDUMMY1;
SELECT INTEGER(2.00) FROM SYSIBM.SYSDUMMY1;
SELECT INTEGER(1.99e0) FROM SYSIBM.SYSDUMMY1;
SELECT INTEGER(2.00e0) FROM SYSIBM.SYSDUMMY1;
SELECT INTEGER(2.00) FROM SYSIBM.SYSDUMMY1;
SELECT INTEGER(1) FROM SYSIBM.SYSDUMMY1;
SELECT DOUBLE(1.00) FROM SYSIBM.SYSDUMMY1;
SELECT CHAR(1.00) FROM SYSIBM.SYSDUMMY1;

-- ROUND()
SELECT ROUND(873.726, 2) FROM SYSIBM.SYSDUMMY1;
SELECT ROUND(873.726, 1) FROM SYSIBM.SYSDUMMY1;
SELECT ROUND(873.726, 0) FROM SYSIBM.SYSDUMMY1;
SELECT ROUND(873.726, -1) FROM SYSIBM.SYSDUMMY1;
SELECT ROUND(873.726, -2) FROM SYSIBM.SYSDUMMY1;
SELECT ROUND(873.726, -3) FROM SYSIBM.SYSDUMMY1;
SELECT ROUND(873.726, -4) FROM SYSIBM.SYSDUMMY1;
