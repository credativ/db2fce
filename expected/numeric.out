-- INTEGER()/INT()
SELECT INTEGER(1.00) FROM SYSIBM.SYSDUMMY1;
ERROR:  syntax error at or near "("
LINE 1: SELECT INTEGER(1.00) FROM SYSIBM.SYSDUMMY1;
                      ^
SELECT INTEGER(1.50) FROM SYSIBM.SYSDUMMY1;
ERROR:  syntax error at or near "("
LINE 1: SELECT INTEGER(1.50) FROM SYSIBM.SYSDUMMY1;
                      ^
SELECT INTEGER(1.99) FROM SYSIBM.SYSDUMMY1;
ERROR:  syntax error at or near "("
LINE 1: SELECT INTEGER(1.99) FROM SYSIBM.SYSDUMMY1;
                      ^
SELECT INTEGER(2.00) FROM SYSIBM.SYSDUMMY1;
ERROR:  syntax error at or near "("
LINE 1: SELECT INTEGER(2.00) FROM SYSIBM.SYSDUMMY1;
                      ^
SELECT INTEGER(1.99e0) FROM SYSIBM.SYSDUMMY1;
ERROR:  syntax error at or near "("
LINE 1: SELECT INTEGER(1.99e0) FROM SYSIBM.SYSDUMMY1;
                      ^
SELECT INTEGER(2.00e0) FROM SYSIBM.SYSDUMMY1;
ERROR:  syntax error at or near "("
LINE 1: SELECT INTEGER(2.00e0) FROM SYSIBM.SYSDUMMY1;
                      ^
SELECT INTEGER(2.00) FROM SYSIBM.SYSDUMMY1;
ERROR:  syntax error at or near "("
LINE 1: SELECT INTEGER(2.00) FROM SYSIBM.SYSDUMMY1;
                      ^
SELECT INTEGER(1) FROM SYSIBM.SYSDUMMY1;
ERROR:  syntax error at or near "("
LINE 1: SELECT INTEGER(1) FROM SYSIBM.SYSDUMMY1;
                      ^
SELECT DOUBLE(1.00) FROM SYSIBM.SYSDUMMY1;
 double 
--------
      1
(1 row)

SELECT CHAR(1.00) FROM SYSIBM.SYSDUMMY1;
ERROR:  syntax error at or near "1.00"
LINE 1: SELECT CHAR(1.00) FROM SYSIBM.SYSDUMMY1;
                    ^
-- ROUND()
SELECT ROUND(873.726, 2) FROM SYSIBM.SYSDUMMY1;
 round  
--------
 873.73
(1 row)

SELECT ROUND(873.726, 1) FROM SYSIBM.SYSDUMMY1;
 round 
-------
 873.7
(1 row)

SELECT ROUND(873.726, 0) FROM SYSIBM.SYSDUMMY1;
 round 
-------
   874
(1 row)

SELECT ROUND(873.726, -1) FROM SYSIBM.SYSDUMMY1;
 round 
-------
   870
(1 row)

SELECT ROUND(873.726, -2) FROM SYSIBM.SYSDUMMY1;
 round 
-------
   900
(1 row)

SELECT ROUND(873.726, -3) FROM SYSIBM.SYSDUMMY1;
 round 
-------
  1000
(1 row)

SELECT ROUND(873.726, -4) FROM SYSIBM.SYSDUMMY1;
 round 
-------
     0
(1 row)

